paragraph/scheduling/log_entry.h

```diff
@@ -0,0 +1,57 @@
+/* Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef PARAGRAPH_SCHEDULING_LOG_ENTRY_H_
+#define PARAGRAPH_SCHEDULING_LOG_ENTRY_H_
+
+#include <string>
+
+#include "paragraph/graph/graph.h"
+
+namespace paragraph {
+
+class LogEntry {
+ public:
+  explicit LogEntry(Instruction* instruction);
+  ~LogEntry() = default;
+
+  // Getters/Setters for instruction timings
+  double GetTimeReady() const;
+  void SetTimeReady(double current_time);
+  double GetTimeStarted() const;
+  void SetTimeStarted(double current_time);
+  double GetTimeFinished() const;
+  void SetTimeFinished(double current_time);
+
+  std::string ToString(const std::string & delimeter = ",") const;
+
+ private:
+  // String identifier for instruction.
+  std::string name_;
```

Comment:
All these members seem to reflect everything that is in the instruction and the instruction FSM. Why not just use a function that takes an InstructionFsm reference and returns a string for the log entry? This is a lot of extra work and extra memory consumption.

Reply:
done